### ISSUE TITLE
Add more `FromMetaItem` impls

### ISSUE THREAD

**Issue by TedDriggs** [TedDriggs/darling#2]
Once specialization lands, it should be possible to add a blanket `impl<T: FromStr> FromMetaItem for T`, but in the meantime it would be useful to have better coverage of the standard library.

**Comment by TedDriggs** [comment]
Update: Support for `darling::Result` is now checked in. This enables fallible parsing. `Vec` is still in progress.

**Comment by TedDriggs** [comment]
Common types are now supported; `Vec` is being deferred in favor of more targeted impls until specialization is available.
